_posts/2020-03-24-dask-developer-workshop.markdown

```diff
@@ -0,0 +1,90 @@
+
+
+### Distributed stability
+We migrated some of our largest data pipelines to Dask in the last months of 2019. For these pipelines, we started to observe significant instability during the computation of the Dask graph. We hadn't seen this issue in any of our previous pipelines running on distributed. Our team contributed several patches upstream in order to resolve these issues on our side. 
+ 
+During the working sessions, a number of conversations took place related to the stability of the distributed scheduler, involving Dask developers and users.
+One idea which came up to increase the overall robustness of distributed was _replication of task results_. The idea is the following: if a worker which is holding a task result is shut down, but the task result is replicated, then we do not need to re-compute parts of the graph again.
+As of the time of this writing, we have not had a chance to actively work on something like this yet, but it is something that we keep in mind.
+
+### Performance and graph optimization
+While monitoring the execution of one of our Dask production pipelines, we observed that the amount of memory consumed by the workers was significantly higher than one would expect in an ideal scenario where tasks are executed in a way which minimizes memory usage of the workers.
+When investigating the Dask.optimization module, we saw that code to optimize memory usage was already there. In practice, however, the graphs are not executed in such an order because of other constraints during execution.
+
+A Dask user at the workshop facing this issue told us that they worked around this by injecting dummy dependencies into Dask graphs. These dependencies acted as "choke-holds" for certain types of tasks, in order to improve the memory usage during execution.
+
+This lack of optimization, also referred to as memory back-pressure, does not only impact memory usage but also increases resource consumption. We would greatly benefit from an implementation which addresses this issue.
+
+## Looking back
+
+### Results from working sessions
+Lucas had the pleasure to collaborate with John Lee on writing down a benchmark to enable [work stealing for tasks with restrictions](https://github.com/Dask/distributed/pull/3069). He also got valuable input from Tom Augspurger on dask internals while debugging a Dask.dataframe bug involving categoricals.
+
+However, arguably the most useful work we did was the implementation of a semaphore in Dask.distributed.
+We need to be able to rate-limit the access of the computation clusters to certain resources such as production databases. Migrating workflows which were still heavily dependent on accessing the database to the distributed scheduler was not possible without a semaphore implementation.
+By working on the semaphore during the workshop, we unblocked a Blue Yonder product team in its migration to Dask.
```

Comment:
This entire sentence is premature + unnecessary, since you say it again two sentences down: "This is something we did not forget;..."

I'd suggest modifying the following three sentences. Something like:

```suggestion
We talked to other workshop attendees who were also interested in a semaphore functionality and during the course of the workshop, actually started implementing it. We finally left DC with a good chunk of the implementation already in place and despite the jetlag, continued working on it once we got back to Karlsruhe. This finally got released...."
```

Reply:
you are omitting `we unblocked a Blue Yonder product team in its migration to Dask` here though

Reply:
Fair point. Updated suggestion:
```suggestion
We talked to other workshop attendees who were also interested in a semaphore functionality and during the course of the workshop, actually started implementing it. We finally left DC with a good chunk of the implementation already in place and despite the jetlag, continued working on it once we got back to Karlsruhe. This finally got [released](https://github.com/Dask/distributed/commit/2129b740c1e3f524e5ba40a0b6a77b239d4c1f94) with distributed 2.14.0 and enabled us to unblock a Blue Yonder product team in its migration to Dask.
```

Reply:
We'll keep it for now